gcloud-java-examples/src/main/java/com/google/gcloud/examples/BigQueryExample.java

```diff
@@ -0,0 +1,686 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.examples;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.gcloud.bigquery.BaseTableInfo;
+import com.google.gcloud.bigquery.BigQuery;
+import com.google.gcloud.bigquery.BigQueryError;
+import com.google.gcloud.bigquery.BigQueryOptions;
+import com.google.gcloud.bigquery.CopyJobInfo;
+import com.google.gcloud.bigquery.DatasetId;
+import com.google.gcloud.bigquery.DatasetInfo;
+import com.google.gcloud.bigquery.ExternalDataConfiguration;
+import com.google.gcloud.bigquery.ExternalTableInfo;
+import com.google.gcloud.bigquery.ExtractJobInfo;
+import com.google.gcloud.bigquery.Field;
+import com.google.gcloud.bigquery.FieldValue;
+import com.google.gcloud.bigquery.FormatOptions;
+import com.google.gcloud.bigquery.JobId;
+import com.google.gcloud.bigquery.JobInfo;
+import com.google.gcloud.bigquery.JobStatus;
+import com.google.gcloud.bigquery.LoadJobInfo;
+import com.google.gcloud.bigquery.QueryRequest;
+import com.google.gcloud.bigquery.QueryResponse;
+import com.google.gcloud.bigquery.Schema;
+import com.google.gcloud.bigquery.TableId;
+import com.google.gcloud.bigquery.TableInfo;
+import com.google.gcloud.bigquery.ViewInfo;
+import com.google.gcloud.spi.BigQueryRpc.Tuple;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An example of using the Google BigQuery.
+ *
+ * <p>This example demonstrates a simple/typical BigQuery usage.
+ *
+ * <p>Steps needed for running the example:
+ * <ol>
+ * <li>login using gcloud SDK - {@code gcloud auth login}.</li>
+ * <li>compile using maven - {@code mvn compile}</li>
+ * <li>run using maven -
+ * {@code mvn exec:java -Dexec.mainClass="com.google.gcloud.examples.BigQueryExample"
+ * -Dexec.args="[<project_id>] list datasets | list tables <dataset> | list jobs |
```

Comment:
Nit: since this is a long list, it may make the javadoc more readable to split this block into lines based on action type or just have each on its own line.